inference-engine/src/cldnn_engine/cldnn_program.h

```diff
@@ -121,12 +122,24 @@ class Program {
     using factory_t = std::function<void(Program&, const std::shared_ptr<ngraph::Node>&)>;
     using factories_map_t = std::map<ngraph::DiscreteTypeInfo, factory_t>;
 
-    template<typename OpType, typename std::enable_if<std::is_base_of<ngraph::Node, OpType>::value, int>::type = 0>
+    template<typename OpType,
+        typename std::enable_if<std::is_base_of<ngraph::Node, OpType>::value && ngraph::HasTypeInfoMember<OpType>::value, int>::type = 0>
     static void RegisterFactory(factory_t func) {
         static std::mutex m;
         std::lock_guard<std::mutex> lock(m);
+        OPENVINO_SUPPRESS_DEPRECATED_START
         if (Program::factories_map.find(OpType::type_info) == Program::factories_map.end())
             Program::factories_map.insert({OpType::type_info, func});
```

Comment:
can we use `get_type_info_static` here as well?

Reply:
I would like to achieve the maximum backward compatibility. If we have a chance to get old operations, I support both cases.